migrations/2018_01_30_222800_create_notification_preferences_table.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Schema\Builder;
+
+return [
+    'up' => function (Builder $schema) {
+        $schema->create('notification_preferences', function (Blueprint $table) {
+            $table->integer('user_id')->unsigned();
+            $table->string('type');
+            $table->string('channel');
+            $table->boolean('enabled')->default(false);
```

Comment:
Do we need an `enabled` column? Doesn't the mere presence of a row in this table indicate that it is enabled?

If there is a good reason to have it, according to the conventions it should be called `is_enabled`

Reply:
We actually need the enabled column, because there would not be a way to distinguish between a user without notification_preferences set yet (and in need for default) and someone who disabled all of them?

Good point about is_enabled.

Reply:
Ah, didn't think of that. Good reason.